ui/apps/ui/src/app/pages/guidelines-page/guideline-detail-page.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Component, OnInit } from '@angular/core';
+import { IResult, ITag } from '@collections/repositories/types';
+import { GuidelinesService } from './guidelines.service';
+import { ActivatedRoute } from '@angular/router';
+import { guidelinesAdapter } from '@collections/data/guidelines/adapter.data';
+import isArray from 'lodash-es/isArray';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import moment from 'moment';
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-guideline-detail-page',
+  templateUrl: './guideline-detail-page.component.html',
+  styleUrls: ['./guideline-detail-page.component.scss'],
+})
+export class GuidelineDetailPageComponent implements OnInit {
+  guideline?: IResult;
```

Comment:
Generally I would suggest more observable oriented approach to this component. Currently you are accessing snapshot of the route, It would have been better to subscribe to the route change and load item based on that. Additionally there is no loading state or anything, also no error redirection - if Id in the route is wrong or does not exist in the db it would be a semi broken page.

Additionally I would suggest having only one object - `model` or `guideline` with all the attributes instead of having them all as properties of the component.

Reply:
What do you think about first refactor iteration?